accumulo/src/main/java/org/vertexium/accumulo/tools/DeleteHistoricalLegacyStreamingPropertyValueData.java

```diff
@@ -0,0 +1,130 @@
+    private final AccumuloGraph graph;
+
+    public DeleteHistoricalLegacyStreamingPropertyValueData(AccumuloGraph graph) {
+        this.graph = graph;
+    }
+
+    public void execute(Options options, Authorizations authorizations) {
+        try {
+            org.apache.accumulo.core.security.Authorizations accumuloAuthorizations = graph.toAccumuloAuthorizations(authorizations);
+            Scanner scanner = graph.getConnector().createScanner(graph.getDataTableName(), accumuloAuthorizations);
+            BatchWriter writer = graph.getConnector().createBatchWriter(
+                    graph.getDataTableName(),
+                    graph.getConfiguration().createBatchWriterConfig()
+            );
+            String lastRowIdPrefix = null;
+            List<Key> rowsToDelete = new ArrayList<>();
+            try {
+                int rowCount = 0;
+                for (Map.Entry<Key, Value> row : scanner) {
+                    if (rowCount % 10000 == 0) {
+                        writer.flush();
+                        LOGGER.debug("looking at row: %s (row count: %d)", row.getKey().getRow().toString(), rowCount);
+                    }
+                    rowCount++;
+                    if (!EMPTY_TEXT.equals(row.getKey().getColumnFamily())) {
+                        continue;
+                    }
+                    if (!EMPTY_TEXT.equals(row.getKey().getColumnQualifier())) {
+                        continue;
+                    }
+
+                    String rowId = row.getKey().getRow().toString();
+                    String[] rowIdParts = rowId.split("" + DataTableRowKey.VALUE_SEPARATOR);
+                    if (rowIdParts.length < 3) {
+                        continue;
+                    }
+
+                    if (lastRowIdPrefix == null || !rowId.startsWith(lastRowIdPrefix)) {
+                        deleteRows(writer, rowsToDelete, options);
+                        rowsToDelete.clear();
+                        lastRowIdPrefix = rowIdParts[0]
+                                + DataTableRowKey.VALUE_SEPARATOR
+                                + rowIdParts[1]
+                                + DataTableRowKey.VALUE_SEPARATOR
+                                + rowIdParts[2];
+                    }
+                    rowsToDelete.add(row.getKey());
+                }
+                deleteRows(writer, rowsToDelete, options);
+            } finally {
+                writer.flush();
+                scanner.close();
+            }
+        } catch (Exception ex) {
+            throw new VertexiumException("Could not delete old SPV data", ex);
+        }
+    }
+
+    private void deleteRows(BatchWriter writer, List<Key> rowsToDelete, Options options) throws MutationsRejectedException {
+        rowsToDelete.sort(Comparator.comparingLong(Key::getTimestamp));
+        int i = 0;
+        for (Key key : rowsToDelete) {
```

Comment:
Not really a problem, but would this loop and counting get any cleaner if you used a while loop and popped the values off the list? I'm thinking something like:
```
while (rowsToDelete.size() > options.getVersionsToKeep()) {
  Key key = rowsToDelete.remove(0);
  LOGGER.debug("deleting row: %s", key.getRow().toString());
  if (!options.isDryRun()) {
  ...
}
```
If you don't want to modify the contents of your parameter, you could put it in a new List first.